docs/cep/source/source-error-handling.md

```diff
@@ -0,0 +1,52 @@
+---
+sidebar_position: 40
+title: Error Handling at Source
+---
+
+When errors are thrown by elements subscribed to the source, the error is propagated up to the source that delivered the event to those stream worker. By default, the error is logged and dropped at the source, but this behavior can be altered by by adding `OnError` property to the corresponding source definition.
+
+### OnError Property
+
+The `OnError` property can help users to capture the error and the associated event, and handle them gracefully by sending them to a fault stream.
+
+The `OnError` property and the required `action` must be specified as below.
+
+```sql
+CREATE SOURCE <stream name> WITH (OnError.action='<action>') (<attribute name> <attribute type>, <attribute name> <attribute type>, ... );
+```
+
+The `action` parameter of the `OnError` property defines the action to be executed during failure scenarios.
+
+The following actions can be specified to `OnError` property to handle erroneous scenarios.
+
+- `STREAM`: Creates a fault stream and redirects the event and the error to it. The created fault stream will have all the attributes defined in the base stream to capture the error causing event, and in addition it also contains `_error` attribute of type `object` to containing the error information. The fault stream can be referred by adding `!` in front of the base stream name as `!<stream name>`.
+
+### Example
+
+Handle errors in `TempStream` by redirecting the errors to a fault stream.
+
+The configuration of `TempStream` stream and `OnError` property is as follows:
+
+```sql
+CREATE STREAM TempStream WITH(OnError.action="STREAM") (deviceID long, roomNo int, temp double;
```

Comment:
@diana-macrometa - The source requires a separate annotation outside the WITH().
Example:
```
@OnError(action='<action>')
CREATE SOURCE <stream name> (<attribute name> <attribute type>, <attribute name> <attribute type>, ... );
```
Ref. https://macrometa.atlassian.net/browse/CEP-486

Reply:
It's within the the ticket `CEP-486`.

Reply:
@diana-macrometa - Please review my changes?

Reply:
Looks good to me!